network/p2p/inspector/validation/control_message_validation_inspector_test.go

```diff
@@ -351,12 +354,66 @@ func TestControlMessageValidationInspector_processInspectRPCReq(t *testing.T) {
 		})
 		require.NoError(t, inspector.processInspectRPCReq(inspectMsgReq))
 	})
+
+	t.Run("inspectRpcPublishMessages should disseminate invalid control message notification when invalid pubsub messages count greater than configured RPCMessageErrorThreshold", func(t *testing.T) {
+		inspector, distributor, _, subscriptions, _, sporkID := inspectorFixture(t)
+		// 5 invalid pubsub messages will force notification dissemination
+		inspector.config.RPCMessageErrorThreshold = 4
+		// create unknown topic
+		unknownTopic := channels.Topic(fmt.Sprintf("%s/%s", unittest.IdentifierFixture(), sporkID)).String()
+		// create malformed topic
+		malformedTopic := channels.Topic("!@#$%^&**((").String()
+		// a topics spork ID is considered invalid if it does not match the current spork ID
+		invalidSporkIDTopic := channels.Topic(fmt.Sprintf("%s/%s", channels.PushBlocks, unittest.IdentifierFixture())).String()
+
+		pubsubMsgs := unittest.GossipSubMessageFixtures(10, fmt.Sprintf("%s/%s", channels.TestNetworkChannel, sporkID))
+		pubsubMsgs = append(pubsubMsgs, []*pubsub_pb.Message{
+			{Topic: &unknownTopic},
+			{Topic: &malformedTopic},
+			{Topic: &malformedTopic},
+			{Topic: &invalidSporkIDTopic},
+			{Topic: &invalidSporkIDTopic},
+		}...)
+		expectedPeerID := peer.ID("peerID987654321")
+		req, err := NewInspectRPCRequest(expectedPeerID, unittest.P2PRPCFixture(unittest.WithPubsubMessages(pubsubMsgs...)))
+		require.NoError(t, err, "failed to get inspect message request")
+		topics := make([]string, len(pubsubMsgs))
+		for i, msg := range pubsubMsgs {
+			topics[i] = *msg.Topic
+		}
+		subscriptions.On("HasSubscription", mock.AnythingOfType("channels.Topic")).Return(true).Run(func(args mock.Arguments) {
+			require.Contains(t, topics, args[0].(channels.Topic).String())
+		})
+		checkNotification := checkNotificationFunc(t, expectedPeerID, p2pmsg.RpcPublishMessage, IsInvalidRpcPublishMessagesErr)
+		distributor.On("Distribute", mock.AnythingOfType("*p2p.InvCtrlMsgNotif")).Return(nil).Once().Run(checkNotification)
+		require.NoError(t, inspector.processInspectRPCReq(req))
+	})
+
+	t.Run("inspectRpcPublishMessages should disseminate invalid control message notification when subscription missing for topic", func(t *testing.T) {
+		inspector, distributor, _, subscriptions, _, sporkID := inspectorFixture(t)
+		// 5 invalid pubsub messages will force notification dissemination
+		inspector.config.RPCMessageErrorThreshold = 4
+		pubsubMsgs := unittest.GossipSubMessageFixtures(10, fmt.Sprintf("%s/%s", channels.TestNetworkChannel, sporkID))
```

Comment:
```suggestion
		pubsubMsgs := unittest.GossipSubMessageFixtures(5, fmt.Sprintf("%s/%s", channels.TestNetworkChannel, sporkID))
```

Reply:
There's no need to create 10 messages (and it's confusing). 5 messages is enough for the test to pass since we're looking for only 4 invalid messages to trigger a notification.